packages/dashboard/src/pages/segments/index.page.tsx

```diff
@@ -49,9 +50,60 @@ export const getServerSideProps: GetServerSideProps<PropsWithInitialState> =
       }
       return result.value;
     });
+    const journeys = await prisma().journey.findMany({
+      where: {
+        workspaceId,
```

Comment:
Can you ensure that only journeys with a `resourceType` of `Declarative` are loaded here? Also can you parallelize this call with `segment.findMany` using `Promise.all`.